### ISSUE TITLE
Publish public Docker image containing the CDK CLI

### ISSUE THREAD

**Issue by clareliguori** [awslabs/aws-cdk#1302]
Some of the newer CI/CD tools out there (Bitbucket Pipelines, GitHub Actions) use Docker images as the environment for each step of the pipeline.  Packaging the CDK CLI as a public Docker image would be a convenient way to integrate releasing CDK-based apps into these types of CI/CD tools.

**Comment by moofish32** [comment]
@clareliguori -- what would you think about also publishing a lambda layers artifact? I've hacked some examples together, but lambda enables a much faster execution and integration with step functions. Yes we could do this with ECS and Fargate now, but that execution was taking about 60 seconds in us-west-2 when I tested a very basic invocation. Lambda is significantly quicker.

**Comment by clareliguori** [comment]
@moofish32 I haven't seen many examples of off-the-shelf or managed CI/CD tools built on functions, so I'd personally probably prioritize container image over Lambda layer (but I'll defer to the CDK team on priority calls).  Sounds like you might be rolling your own with Step Functions?  Neat!

**Comment by clareliguori** [comment]
I don't think this is resolved.  The ask here is for a public Docker image, for example published into the amazon DockerHub namespace.  Also, the current Dockerfile is not suitable for publishing publicly: it's multiple GBs with cruft like devDependencies and the .git folder, whereas the GitHub release archives are < 200 MB.

**Comment by ScottBrenner** [comment]
Kind of related -

I made a GitHub Action for the CDK https://github.com/marketplace/actions/aws-cdk-action
.. then I realized that for people to be able to use it, it needs to have _all_ the CDK packages/dependencies installed in it, like `@aws-cdk/aws-s3` etc .. and maybe even compile the code too ..

Guess it's not feasible? :sob:

**Comment by cristianrat** [comment]
@ScottBrenner Very easy to build your own. I've done just that with only the packages I need.
There are also a couple people who made them public on docker hub.

**Comment by stefanx-amazon** [comment]
While I agree that this is easy to build for anybody, wouldn't it be desirable to have an official CDK Docker image available on Docker Hub and also e.g. https://gallery.ecr.aws/aws-cdk/aws-cdk that all CDK users can use and rely on rather than everybody building their own CDK Docker images?
If that existed, we could also update e.g. [the documentation for CDK Pipelines](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html) to use the official CDK Docker image rather than including `npm install -g cdk`, which would speed up pipelines of users that don't bother creating their own CDK Docker image quite a bit.

**Comment by gliptak** [comment]
Published my Python construct image at https://github.com/gliptak/aws-cdk-python

#14821

**Comment by RajasGujarathi** [comment]
This is the need of the hour. An official docker image will be beneficial with CI/CD tools like GitLab and others mentioned above.
